workspaces/marketplace/plugins/marketplace/src/api/MarketplaceBackendClient.tsx

```diff
@@ -39,9 +42,12 @@ export class MarketplaceBackendClient implements MarketplaceApi {
     this.fetchApi = options.fetchApi;
   }
 
-  async getPlugins(): Promise<MarketplacePlugin[]> {
+  async getPlugins(
+    request?: GetPluginsRequest,
+  ): Promise<MarketplacePluginWithPageInfo> {
     const baseUrl = await this.discoveryApi.getBaseUrl('marketplace');
-    const url = `${baseUrl}/plugins`;
+    const params = convertGetPluginRequestToSearchParams(request);
+    const url = `${baseUrl}/plugins?${params.toString()}`;
```

Comment:
Let us do the extra step and don't append the `?`  if no params is needed.

```suggestion
    const params = convertGetPluginRequestToSearchParams(request);
    const query = params.toString();
    const url = `${baseUrl}/plugins${query ? '?' : ''}${query}`;
```